### ISSUE TITLE
For Germany the numbers don't match the official numbers

### ISSUE THREAD

**Issue by asmaier** [CSSEGISandData/COVID-19#826]
As of today for Germany the official numbers of total confirmed cases are 4838 (according to RKI, the german CDC): https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Fallzahlen.html

However the board https://www.arcgis.com/apps/opsdashboard/index.html#/bda7594740fd40299423467b48e9ecf6
shows a number of 5813.

What are your sources for Germany and how do you explain the difference of nearly 1000 cases ?

**Comment by foursixnine** [comment]
See page 2 of the Situationsberichte (It's also in English), and then read that foot note:  https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Situationsberichte/2020-03-15-en.pdf?__blob=publicationFile

And also read the terms of use of this repo/board, and keep in mind that it is not something that itś updated in real time, there is some lag.

**Comment by b2m9** [comment]
> See page 2 of the Situationsberichte (It's also in English), and then read that foot note: https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Situationsberichte/2020-03-15-en.pdf?__blob=publicationFile

I actually assumed they just got a number wrong, so I was curious about your link. Yet, I can't see what you want to point out. RKI reports 4,838 manually reported cases and 4,195 electronically reported cases. What am I missing here?

Anyhow, I'm sure Johns Hopkins will correct the numbers once they are back at work.

**Comment by neurofreakpb** [comment]
From the German CDC (RKI) [Situationsberichte](https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Situationsberichte/2020-03-15-en.pdf?__blob=publicationFile), page 2
> A total of 4,838 (+1,043) laboratory-confirmed cases of coronavirus disease 2019 (COVID-19) have been detected in Germany since 27/01/2020, of which 4,195 were electronically reported to and validated at the RKI. So far, 12 (+4) deaths related to COVID-19 diseases were reported.

This gives a total of 4838+1043=5881 cases. This is the official number. The number reported in this repo is 5883.

What the number in the parenthesis represents is not explained even in the RKI document, so it'd be better if only one of these is used in this repo: manually reported cases (4838), or electronically reported cases (4195) for the sake of clarity and consistency in the future as well :)

**Comment by b2m9** [comment]
> This gives a total of 4838+1043=5881 cases. This is the official number. The number reported in this repo is 5883.

> What the number in the parenthesis (i.e. 1043) represents is not explained even in the RKI document

Nope, they clearly state at the beginning that the blue numbers are the differences compared to the previous report. So it is 4838 in total as of 15 March, which is 1043 cases more than the previous report. Hence, the big fat number 4838 of total cases on the cover of the report.

From the report:
> –Changes since the last report have been marked blue in the text

**Comment by neurofreakpb** [comment]
> Because they clearly state at the beginning that the blue numbers are the differences compared to the previous report.
That's right. Thank you for pointing it out.

1043 is, therefore, the increase in the number of cases from 03/14 to 03/15. This means that adding 1043 and 4838 to get a total of 5881 doesn't make sense! Thus the total number of cases remains 4838 (or 4195 if one considers electronically submitted cases only) and so does the issue here.

**Comment by asmaier** [comment]
As of today the official number according to RKI is 6012 (see also https://qap.ecdc.europa.eu/public/extensions/COVID-19/COVID-19.html) . WHO ist one day behind: 4838 (https://experience.arcgis.com/experience/685d0ace521648f8a5beeeee1b9125cd). 

The German Wikipedia entry (https://de.wikipedia.org/wiki/COVID-19-Epidemie_in_Deutschland) lists the numbers from RKI: 6012
But the English Wikipedia entry (https://en.wikipedia.org/wiki/2020_coronavirus_pandemic_in_Germany) 
takes it numbers from https://interaktiv.morgenpost.de/corona-virus-karte-infektionen-deutschland-weltweit/ (which claims to also parse information from the german federal states health care ministries e.g. https://www.mags.nrw/coronavirus-fallzahlen-nrw), which shows a much higher number: 7.974

However John Hopkins list 7689 cases, which is again another number. And it is really unclear to me, where you get it from.

**Comment by jgehrcke** [comment]
Here I explain why the RKI numbers are sometimes affected by a 1-2 day delay from what the individual ministries of health in Germany actually publish: https://gehrcke.de/2020/03/covid-19-http-api-for-german-case-numbers/

**Comment by asmaier** [comment]
By the way, the zeit.de (a german newspaper) data can be retrieved from this URL: https://interactive.zeit.de/cronjobs/2020/corona/data.json
This is probably the most up-to-date data for Germany that can be retrieved via an open API.

**Comment by jgehrcke** [comment]
> This is probably the most up-to-date data for Germany that can be retrieved via an open API.

It's not "open", though, as it is an implementation detail in ZEIT ONLINE's architecture. We might or might not be allowed to use it, and they might change the implementation details at any time.

**Comment by jgehrcke** [comment]
I haven't really documented that yet elsewhere (doing that tomorrow), but here's what you can use:

### Current snapshot for all of Germany (no time series)

```
$ curl -s https://covid19-germany.appspot.com/now | jq
{
  "current_totals": {
    "cases": 12223,
    "deaths": 31,
    "recovered": 99,
    "tested": "unknown"
  },
  "meta": {
    "contact": "Dr. Jan-Philip Gehrcke, jgehrcke@googlemail.com",
    "source": "ZEIT ONLINE (aggregated data from individual ministries of health in Germany)",
    "time_source_last_consulted_iso8601": "2020-03-19T03:47:01+00:00",
    "time_source_last_updated_iso8601": "2020-03-18T22:11:00+01:00"
  }
}
```

### For the history (time series) information for each German state:

Use `https://covid19-germany.appspot.com/timeseries/<state>/<metric>`:

For `<state>` use the [ISO_3166](https://en.wikipedia.org/wiki/ISO_3166-2:DE) code, for `<metric>` use `cases` or `deaths`.

For example the number of confirmed cases over time for Bayern (Bavaria):
```
$ curl -s https://covid19-germany.appspot.com/timeseries/DE-BY/cases | jq
{
  "data": {
    "2020-03-10T12:00:00+01:00": "314",
    "2020-03-11T12:00:00+01:00": "366",
    "2020-03-12T12:00:00+01:00": "500",
    "2020-03-13T12:00:00+01:00": "558",
    "2020-03-14T12:00:00+01:00": "681",
    "2020-03-15T12:00:00+01:00": "886",
    "2020-03-16T12:00:00+01:00": "1067",
    "2020-03-17T21:00:00+01:00": "1352",
    "2020-03-18T23:00:00+01:00": "1798"
  },
  "meta": {
    "info": "https://gehrcke.de/2020/03/covid-19-http-api-german-states-timeseries",
    "source": "Official numbers published by public health offices (Gesundheitsaemter) in Germany"
  }
}

```

**Comment by asmaier** [comment]
@jgehrcke Are you affiliated with zeit.de ? I don't understand why you say the URL https://interactive.zeit.de/cronjobs/2020/corona/data.json is not open when it clearly is. If they would not want everybody to access that API they should hide it behind a Firewall, or put a password or demand an API-Token. As of now this URL seems to be accessible by everyone in the world.

**Comment by jgehrcke** [comment]
> @jgehrcke Are you affiliated with zeit.de ? I don't understand why you say the URL https://interactive.zeit.de/cronjobs/2020/corona/data.json is not open

Hey @asmaier -- I am not affiliated with ZEIT ONLINE. The URL it is publicly accessible, yes. But that's about it. And of course I am not telling that you should not use it. I was merely giving a heads-up, maybe a kind warning: an "open API" is something vastly different. An open API by more modern standards has the clear declaration of purpose and intent ("this is here for _you_ to use!"), a clear interface specification, maintains interface stability, and uses and a data format that is easily usable from tools and tooling. I looked into their JSON document format quite a bit and of course parsing that is a bit of a pain.

Again, I have a strong opinion about what an "open API" is and maybe what not, but I am certainly not telling you what to do, sorry if I raised that impression :-).

**Comment by asmaier** [comment]
So I figured out a similar "Open API" as for ZEIT ONLINE for Berliner Morgenpost:

Historical case data for Germany, Europe and the World:

- https://interaktiv.morgenpost.de/corona-virus-karte-infektionen-deutschland-weltweit/data/Coronavirus.timeline.v2.csv

Latest data for countries and federal states in Germany:
- https://interaktiv.morgenpost.de/corona-virus-karte-infektionen-deutschland-weltweit/data/Coronavirus.current.v2.csv

Regional data for Germany:
- https://interaktiv.morgenpost.de/corona-virus-karte-infektionen-deutschland-weltweit/data/Coronavirus.regio.v2.csv

The nice thing is, they store a source url for each data point.

**Comment by jgehrcke** [comment]
@asmaier these are indeed great data sources, look well-researched. With sources and timestamps.

In https://github.com/jgehrcke/covid-19-germany-gae I have changed the `/now` endpoint (https://covid19-germany.appspot.com/now) to be based on multiple sources now, including Berliner Morgenpost, and to report the more recent case count. Maybe that is useful. Cheers!
